anta/settings.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023-2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""Settings for ANTA."""
+
+from __future__ import annotations
+
+import logging
+import os
+import sys
+from enum import Enum
+from typing import Any
+
+from pydantic import Field, PositiveInt
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+logger = logging.getLogger(__name__)
+
+# Default value for the maximum number of concurrent tests in the event loop
+DEFAULT_MAX_CONCURRENCY = 10000
+
+# Default value for the maximum number of open file descriptors for the ANTA process
+DEFAULT_NOFILE = 16384
+
+# Default value for the test scheduling strategy
+DEFAULT_SCHEDULING_STRATEGY = "round-robin"
+
+# Default value for the number of tests to schedule per device when using the DEVICE_BY_COUNT scheduling strategy
+DEFAULT_SCHEDULING_TESTS_PER_DEVICE = 100
+
+
+class AntaRunnerSchedulingStrategy(str, Enum):
+    """Enum for the test scheduling strategies available in the ANTA runner.
+
+    * ROUND_ROBIN: Distribute tests across devices in a round-robin fashion.
+    * DEVICE_BY_DEVICE: Run all tests on a single device before moving to the next.
+    * DEVICE_BY_COUNT: Run all tests on a single device for a specified count before moving to the next.
+
+    NOTE: This could be updated to StrEnum when Python 3.11 is the minimum supported version in ANTA.
+    """
+
+    ROUND_ROBIN = "round-robin"
+    DEVICE_BY_DEVICE = "device-by-device"
+    DEVICE_BY_COUNT = "device-by-count"
+
+    def __str__(self) -> str:
+        """Override the __str__ method to return the value of the Enum, mimicking the behavior of StrEnum."""
+        return self.value
+
+
+class AntaRunnerSettings(BaseSettings):
```

Comment:
not for this PR but probably we should think about making a config file down the line (though I hate interaction between config file/ CLI / env variable :p)